### PR TITLE
Simplify mkmf-rice

### PR DESCRIPTION
**Pull request by ankane**
This PR simplifies `mkmf-rice` by:

- Removing code for Ruby < 2.7
- Using `match?` (added in Ruby 2.4) rather than `!match.nil?`

### CODE REVIEW

File: lib/mkmf-rice.rb [hunk 1]
```diff
@@ -1,93 +1,8 @@
...
-      def link_command(ldflags, *opts)
-        conf = link_config(ldflags, *opts)
-        RbConfig::expand(TRY_LINK_CXX.dup, conf)
-      end
-    end
-  end
-end
+IS_MSWIN = RbConfig::CONFIG['host_os'].match?(/mswin/)
+IS_MINGW = RbConfig::CONFIG['host_os'].match?(/mingw/)
+IS_DARWIN = RbConfig::CONFIG['host_os'].match?(/darwin/)
```
Comment (cfis) [inline lib/mkmf-rice.rb:5]: I noticed this today - maybe we can use these:

https://github.com/ruby/ruby/blob/master/lib/mkmf.rb#L139

```
 $mswin = /mswin/ =~ RUBY_PLATFORM
  $mingw = /mingw/ =~ RUBY_PLATFORM
  $cygwin = /cygwin/ =~ RUBY_PLATFORM
```

Nothing for Darwin though.

File: lib/mkmf-rice.rb [hunk 2]
```diff
@@ -1,93 +1,8 @@
...
 
 # The cpp_command is not overwritten in the experimental mkmf C++ support.
 # See https://bugs.ruby-lang.org/issues/17578
```
Comment (cfis) [inline lib/mkmf-rice.rb:8]: If this still an issue, we should comment on my original bug and see if we can get this fixed.

**Comment by cfis** [comment]
I was actually looking at this code today and thinking about removing it.

I'll merge this PR, but see my comments for a possible additional change.

**Comment by ankane** [comment]
Thanks, addressed the comments in #212

**Pull request merged**